modules/ballerina-core/src/main/java/org/ballerinalang/bre/BLangExecutor.java

```diff
@@ -1073,6 +1073,10 @@ private void assignValueToArrayMapAccessExpr(BValue rValue, ArrayMapAccessExpr l
         if (!(accessExpr.getRExpr().getType() == BTypes.typeMap)) {
             BArray arrayVal = (BArray) accessExpr.getRExpr().execute(this);
 
+            if (arrayVal == null) {
+                throw new BallerinaException("field " + accessExpr.getSymbolName() + " is not initialized");
```

Comment:
Shall we reword this to "variable 'x' is null" as in [1] to maintain the consistency in the error messages?

[1] https://github.com/ballerinalang/ballerina/blob/master/modules/ballerina-core/src/main/java/org/ballerinalang/bre/BLangExecutor.java#L734